src/videojs-contrib-hls.js

```diff
@@ -674,12 +674,23 @@ const HlsSourceHandler = function(mode) {
 
       let settings = videojs.mergeOptions(options, {hls: {mode}});
 
+      let previousBeforeRequest;
+
+      if (tech.hls && tech.hls.xhr && tech.hls.xhr.beforeRequest) {
+        previousBeforeRequest = tech.hls.xhr.beforeRequest;
```

Comment:
Could change this to `previousBeforeRequest = tech.hls.xhr.beforeRequest || videojs.Hls.xhr.beforeRequest;` This could remove the need for the if statements below to instead just always set `tech.hls.xhr.beforeRequest = previousBeforeRequest;`. However, the code in its current state may be more understandable, so I'd only consider this comment if you think its an improvement there